utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -0,0 +1,281 @@
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    assertThat(performanceMetricsUtilsImpl.getUsedStorage()).isEqualTo(expectedStorageValue)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getTotalStorageUsageTier_returnsCorrectStorageUsageTier() {
+    context.openFileOutput(TEST_FILE_NAME, Context.MODE_PRIVATE).use {
+      it.write(TEST_FILE_CONTENT.toByteArray())
+    }
+    val cacheFile = File.createTempFile(TEST_APP_PATH, null, context.cacheDir)
+    val expectedStorageValue = TEST_FILE_CONTENT.length + cacheFile.length()
+
+    val expectedStorageTierValue = when (expectedStorageValue / (1024 * 1024 * 1024)) {
+      in 0L until 32L -> OppiaMetricLog.StorageTier.LOW_STORAGE
+      in 32L..64L -> OppiaMetricLog.StorageTier.MEDIUM_STORAGE
+      else -> OppiaMetricLog.StorageTier.HIGH_STORAGE
+    }
+
+    assertThat(performanceMetricsUtilsImpl.getDeviceStorageTier())
+      .isEqualTo(expectedStorageTierValue)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getBytesSent_returnsCorrectAmountOfNetworkBytesSent() {
+    val expectedNetworkBytesSent = TrafficStats.getUidTxBytes(
+      ApplicationProvider.getApplicationContext<Application>().applicationInfo.uid
+    )
+
+    assertThat(performanceMetricsUtilsImpl.getTotalSentBytes())
+      .isEqualTo(expectedNetworkBytesSent)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_getBytesReceived_returnsCorrectAmountOfNetworkBytesReceived() {
+    val expectedNetworkBytesReceived = TrafficStats.getUidRxBytes(
+      ApplicationProvider.getApplicationContext<Application>().applicationInfo.uid
+    )
+
+    assertThat(performanceMetricsUtilsImpl.getTotalReceivedBytes())
+      .isEqualTo(expectedNetworkBytesReceived)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_setAppProcesses_getMemoryUsage_returnsCorrectMemoryUsage() {
+    val activityManager: ActivityManager =
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    val process1 = ActivityManager.RunningAppProcessInfo().apply {
+      this.pid = TEST_PID
+      this.importanceReasonComponent = ComponentName("com.robolectric", "process 1")
+    }
+    val shadowActivityManager: ShadowActivityManager = shadowOf(activityManager)
+    shadowActivityManager.setProcesses(listOf(process1))
+    val processMemoryInfo = activityManager.getProcessMemoryInfo(arrayOf(TEST_PID).toIntArray())
+    val totalPssUsedTest = processMemoryInfo?.get(0)?.totalPss?.toLong() ?: 0L
+
+    assertThat(performanceMetricsUtilsImpl.getTotalPssUsed()).isEqualTo(totalPssUsedTest)
+  }
+
+  @Test
+  fun testPerformanceMetricsUtils_removeCurrentApp_installTestApp_returnsCorrectApkSize() {
+    val application: Application = ApplicationProvider.getApplicationContext()
+    val applicationInfo = ApplicationInfo()
+    val testApkSize = (File(TEST_APP_PATH).length() / 1024)
```

Comment:
Isn't this just 0? It doesn't seem like the file is actually getting written. Could we write real content and then have an assertion for the exact byte value?

Reply:
done.